### PR TITLE
os/bluestore: ignore 0x2000~2000 extent oddity from luminous upgrade

### PR DESCRIPTION
**Pull request by liewegas**
Luminous does a block_size granularity freelist, and assumes that
0\~ROUND_UP_TO(SUPER_RESERVED,block_size) is used.  Current master uses
min_alloc_size granularity and changes that assumption to
0\~ROUND_UP_TO(SUPER_RESERVED,min_alloc_size).  That means if master
fsck's a luminous-created bluestore, it will think 0x2000~2000 is used
(current baked-in min_alloc_size-based assumption) but the old freelist
says it is free (old mkfs assumption).  The disparity is harmless since
the extent is below min_alloc_size, so ignore it.

Fixes: http://tracker.ceph.com/issues/21408
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by liewegas** [comment]
@yuriw this should make your upgrade/luminous-x error go away!

**Comment by varadakari** [comment]
Missed them in the initial review.

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -6011,10 +6011,22 @@ int BlueStore::_fsck(bool deep, bool repair)
         }
       );
       if (intersects) {
-	derr << "fsck error: free extent 0x" << std::hex << offset
-	     << "~" << length << std::dec
-	     << " intersects allocated blocks" << dendl;
-	++errors;
+	if (offset == SUPER_RESERVED &&
+	    length == min_alloc_size - SUPER_RESERVED) {
```
Comment (xiexingguo) [inline src/os/bluestore/BlueStore.cc]: We'd better confirm that min_alloc_size is greater than SUPER_RESERVED here since min_alloc_size is configurable

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-09-22_14:27:10-rados-wip-sage-testing-2017-09-21-2230-distro-basic-smithi/

**Pull request merged**